azuredevops/internal/service/build/resource_resource_authorization.go

```diff
@@ -120,55 +154,64 @@ func resourceResourceAuthorizationDelete(d *schema.ResourceData, m interface{})
 	// because the resource to delete might have had this parameter set to true, we overwrite it
 	authorizedResource.Authorized = converter.Bool(false)
 
-	err = sendAuthorizedResourceToAPI(clients, authorizedResource, projectID)
+	err = sendAuthorizedResourceToAPI(clients, authorizedResource, projectID, definitionID)
 	if err != nil {
 		return fmt.Errorf(msgErrorFailedResourceDelete, err)
 	}
 
-	return err
+	return nil
 }
 
 func resourceResourceAuthorizationUpdate(d *schema.ResourceData, m interface{}) error {
 	clients := m.(*client.AggregatedClient)
-	authorizedResource, projectID, err := expandAuthorizedResource(d)
+	authorizedResource, projectID, definitionID, err := expandAuthorizedResource(d)
 	if err != nil {
 		return fmt.Errorf(msgErrorFailedResourceUpdate, err)
 	}
 
-	err = sendAuthorizedResourceToAPI(clients, authorizedResource, projectID)
+	err = sendAuthorizedResourceToAPI(clients, authorizedResource, projectID, definitionID)
 	if err != nil {
 		return fmt.Errorf(msgErrorFailedResourceUpdate, err)
 	}
 
 	return resourceResourceAuthorizationRead(d, m)
 }
 
-func flattenAuthorizedResource(d *schema.ResourceData, authorizedResource *build.DefinitionResourceReference, projectID string) {
+func flattenAuthorizedResource(d *schema.ResourceData, authorizedResource *build.DefinitionResourceReference, projectID string, definitionID int) {
 	d.SetId(*authorizedResource.Id)
 	d.Set("resource_id", *authorizedResource.Id)
```

Comment:
Can you remove all pointer dereferences when `d.Set` is called, because the `Set` function is very well capable of handling `nil`pointers. This will also minimize the possibility of a `panic` when the API might change an return an empty `nil`value for a computed or optional field.

https://github.com/hashicorp/terraform-plugin-sdk/blob/621b20845686c999865f87682650d38f191b5483/helper/schema/resource_data.go#L164

Reply:
Done for this, not doing everything, you maniac :D